riscv: telink: load DAC key/cert from flash zone.

pulin1103 opened the pull request.

fengtai-telink commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,60 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+
+#if CHIP_DEVICE_SECURE_PROGRAMMING
+
+#if defined(CONFIG_SOC_RISCV_TELINK_B92)
```
是不是可以改成 #ifdef CONFIG_SOC_RISCV_TELINK_B92

liucoldstar replied:
可以统一到#ifdef

fengtai-telink commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,60 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+#include "aes.h"
+#endif
+
+#if defined(CONFIG_SOC_RISCV_TELINK_TL321X)
```
#if defined -> #ifdef

fengtai-telink commented on config/telink/chip-module/Kconfig:
```diff
@@ -254,3 +254,8 @@ config CHIP_USE_MARS_SENSOR
+config SECURE_PROGRAMMING
+	bool "Enable reading DAC and keys from the DAC partition"
+    depends on CHIP_FACTORY_DATA
+    default n
```
代码没对齐

fengtai-telink replied:
还有就是，我记得海文说过，Kconfig里面不要Enable xxx，CI会不过，直接Read DAC and keys from the DAC partition会不会简单点

andy-liu-telink replied:
可以看CI结果确认一下。

fengtai-telink replied:
刚和海文确认了下，matter这边没这个问题，那应该没什么问题

andy-liu-telink commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,60 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+    {
+        return false;
+    }
+    if (efuse_get_chip_id(chip_id))
```
efuse_get_chip_id在B92和TL321x中的返回值不同的，处理逻辑刚好相反。可以参考金邈的这个PR里面的处理方式：https://github.com/telink-semi/mcuboot/pull/3

liucoldstar commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,60 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+    }
+
+    size_t dac_cert_len;
+    dac_cert_len = buffer[100];
```
100用宏表示一下吧。 比如
static constexpr uint32_t kDacContentsOffset = 100;

liucoldstar commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,60 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+
+    size_t dac_cert_len;
+    dac_cert_len = buffer[100];
+    dac_cert_len |= (uint16_t) buffer[101] << 8;
```
按照上面的处理方式，这里可以101可以替换成kDacContentsOffset + 1

liucoldstar commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,60 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+    {
+        return false;
+    }
+    factoryData->dac_cert.data = buffer + 102;
```
kDacContentsOffset + 2

Commit:
riscv: telink: load DAC key/cert from flash zone.

Signed-off-by: Zhenghuan Zhang <zhenghuan.zhang@telink-semi.com>

Commit:
telink: extend and update mfg_tool

- add plaintext vid, pid & discriminator at the beginning.
- add optional argument --secure-programming-verification and --chip-id to
generate factory data and independent DAC.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>
Signed-off-by: Zhenghuan Zhang <zhenghuan.zhang@telink-semi.com>

Commit:
riscv: telink: fix error format

Signed-off-by: Zhenghuan Zhang <zhenghuan.zhang@telink-semi.com>

damien0x0023 committed:
telink: update factory data and dac

- modify mfg_tool code format.
- add merge_factorydata_dac.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

damien0x0023 committed:
telink update mfg_tool and merge_factorydata_dac

to fix code style issues.

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>

andy-liu-telink commented on src/platform/telink/FactoryDataParser.c:
```diff
@@ -181,3 +187,70 @@ bool ParseFactoryData(uint8_t * buffer, uint16_t bufferSize, struct FactoryData
+    }
+    else
+    {
+        LOG_ERR("Private key decryption failed.");
```
这个错误信息应该改成“Failed to retrieve chip ID.”

damien0x0023 replied:
@pulin1103 How about we add this change to my next Pull Request?
